examples/knowledge-base/pages/api/search.ts

```diff
@@ -0,0 +1,76 @@
+import lunr from "lunr";
+import type { NextApiRequest, NextApiResponse } from "next";
+import fs from "node:fs/promises";
+import { documentToPlainTextString } from "@contentful/rich-text-plain-text-renderer";
+import dotenv from "dotenv-flow";
+
+dotenv.config();
+
+import { buildSearchIndex } from "../../lib/search";
+import { getSingleArticleBySlug } from "../../lib/api";
+
+const truncateContent = async (found: lunr.Index.Result) => {
+  const key = Object.keys(found.matchData?.metadata).find((key) => {
+    // console.log(found.matchData.metadata[key]);
+    return found.matchData.metadata[key].content?.position;
+  });
+
+  const articleSlug = found.ref;
+  const contentfulResult = await getSingleArticleBySlug(articleSlug);
+  const text = documentToPlainTextString(contentfulResult.body.json);
+
+  if (!key) return text;
+
+  const [index, length] = found.matchData.metadata[key].content.position[0];
+  const startIndex = Math.max(0, index - 15);
+  const truncatedContent = text.slice(startIndex, startIndex + length + 50);
+  let content;
```

Comment:
```suggestion
  let content = truncatedContent;
```
Otherwise it gets undefined if the next if is falsy

Sorry I missed this earlier 😞